src/Database/Media/Utilities.ts

```diff
@@ -3,7 +3,8 @@ export function getIdFromOutpoint(outpoint: string) {
 }
 
 export function getOutpointFromId(id: string) {
-  const outpoint = id.split("");
-  outpoint[id.length - 2] = ":";
-  return outpoint.join("");
+  return id
+    .split("")
+    .map((x, i, arr) => (i === arr.length - 2 ? ":" : x))
+    .join("");
```

Comment:
This seems to now map over every character in the string instead of a constant replace?

Reply:
yes, for all characters it will return the original char, for that specific index it return the `:`

Reply:
```javascript
const old = (str) => {
  const outpoint = str.split("");
  outpoint[str.length - 2] = ":";
  return outpoint.join("");
};

const test = (str) =>
  str
    .split("")
    .map((x, i, arr) => (i === arr.length - 2 ? ":" : x))
    .join("");

console.log(old("foobar"));  // foob:r
console.log(test("foobar")); // foob:r
```